packages/core/src/metadata/MetadataDiscovery.ts

```diff
@@ -1496,6 +1494,22 @@ export class MetadataDiscovery {
     return this.platform.getMappedType(t);
   }
 
+  private getPrefix(prop: EntityProperty, parent: EntityProperty | null): string {
+    const { embeddedPath = [], fieldNames, prefix = true, prefixBehavior } = prop;
+
+    if (prefix === true) {
+      return (embeddedPath.length ? embeddedPath.join('_') : fieldNames[0]) + '_';
+    }
+
+    const prefixParent = parent ? this.getPrefix(parent, null) : '';
+    if (prefix === false) {
+      return prefixParent;
+    }
+
+    const behavior = prefixBehavior ?? this.config.getAll().embeddable.prefixBehavior;
```

Comment:
i also renamed the key here to `embeddables` instead of `embeddable`

```suggestion
    const behavior = prefixBehavior ?? this.config.get('embeddables').prefixBehavior;
```